packages/interface/src/components/fullscreen-mode/index.js

```diff
@@ -1,51 +1,48 @@
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
+import { useEffect, useState } from '@wordpress/element';
 
-export class FullscreenMode extends Component {
-	componentDidMount() {
-		this.isSticky = false;
-		this.sync();
+export const FullscreenMode = ( { isActive } ) => {
+	const [ isSticky, setIsSticky ] = useState( false );
 
+	// componentDidMount
+	useEffect( () => {
 		// `is-fullscreen-mode` is set in PHP as a body class by Gutenberg, and this causes
 		// `sticky-menu` to be applied by WordPress and prevents the admin menu being scrolled
 		// even if `is-fullscreen-mode` is then removed. Let's remove `sticky-menu` here as
 		// a consequence of the FullscreenMode setup
 		if ( document.body.classList.contains( 'sticky-menu' ) ) {
-			this.isSticky = true;
+			setIsSticky( true );
 			document.body.classList.remove( 'sticky-menu' );
 		}
-	}
-
-	componentWillUnmount() {
-		if ( this.isSticky ) {
-			document.body.classList.add( 'sticky-menu' );
-		}
-
-		if ( this.props.isActive ) {
-			document.body.classList.remove( 'is-fullscreen-mode' );
-		}
-	}
-
-	componentDidUpdate( prevProps ) {
-		if ( this.props.isActive !== prevProps.isActive ) {
-			this.sync();
-		}
-	}
-
-	sync() {
-		const { isActive } = this.props;
+	}, [ setIsSticky ] );
+
+	// componentWillUnmount
+	useEffect( () => {
+		return () => {
+			if ( isSticky ) {
+				document.body.classList.add( 'sticky-menu' );
+			}
+		};
+	}, [ isSticky ] );
+
+	useEffect( () => {
+		return () => {
+			if ( isActive ) {
+				document.body.classList.remove( 'is-fullscreen-mode' );
+			}
+		};
+	}, [ isActive ] );
+
+	// componentDidUpdate
+	useEffect( () => {
 		if ( isActive ) {
 			document.body.classList.add( 'is-fullscreen-mode' );
 		} else {
 			document.body.classList.remove( 'is-fullscreen-mode' );
 		}
-	}
-
-	render() {
-		return null;
-	}
-}
+	}, [ isActive ] );
```

Comment:
I think there are some issues with the logic here, I see:

 - isSticky is not a state in the previous implementation and changing its value shouldn't cause an effect to run, basically it needs to be stored in a ref ( `useRef` )
 - We can probably combine the effects that have `isActive` as a dependency

Reply:
@youknowriad Thank you for the review. I modified useStatus to useRef.